sentry-core/src/main/java/io/sentry/core/SentryClient.java

```diff
@@ -48,6 +48,30 @@ public SentryId captureEvent(SentryEvent event, @Nullable Scope scope) {
       event.setEnvironment(options.getEnvironment());
     }
 
+    if (scope != null) {
+      if (event.getTransaction() == null) {
+        event.setTransaction(scope.getTransaction());
+      }
+      if (event.getUser() == null) {
+        event.setUser(scope.getUser());
+      }
+      if (event.getFingerprint() == null) {
+        event.setFingerprint(scope.getFingerprint());
+      }
+      if (event.getBreadcrumbs() == null) {
```

Comment:
The ones above look good but for Tags, Extra and Breadcrumbs I think it should look more like this:

```java
if (event.getBreadcrumbs() == null) {
  // Copy breadcrumbs (which are immutable) into a new list
  event.setBreadcrumbs(new ArrayList<>(scope.getBreadcrumbs()));
} else {
  // append items to event breadcrumb list
  List<Breadcrumb> breadcrumbs = event.getBreadcrumbs();
  for (Breadcrumb scopeBreadcrumbs: scope.getBreadcrumbs()) {
    breadcrumbs.add(scopeBreadcrumbs);
  }
}
```